components/ui/Feed/FeedSortButton.tsx

```diff
@@ -0,0 +1,62 @@
+import { useTheme } from "native-base";
+import HeaderIconButton from "../buttons/HeaderIconButton";
+import { useActionSheet } from "@expo/react-native-action-sheet";
+import { UseFeed } from "../../hooks/feeds/useFeed";
+import { SortType } from "lemmy-js-client";
+import {
+  IconBolt,
+  IconCalendar,
+  IconClockHour4,
+  IconFlame,
+  IconMessage,
+} from "tabler-icons-react-native";
+import { IconCalendarWeek } from "../customIcons";
+
+type SortOption = [key: SortType, display: string];
+
+const sortOptions = [
+  ["TopDay", "Top Day"],
+  ["TopWeek", "Top Week"],
+  ["Hot", "Hot"],
+  ["Active", "Active"],
+  ["New", "New"],
+  ["MostComments", "Most Comments"],
+] satisfies SortOption[];
+
+interface Props {
+  feed: UseFeed;
+  onSortUpdate?: (key: SortType) => void;
+}
+
+export const FeedSortButton = ({ feed, onSortUpdate }: Props) => {
+  const theme = useTheme();
+  const { showActionSheetWithOptions } = useActionSheet();
+
+  const onPress = () => {
+    const cancelButtonIndex = sortOptions.length;
+
+    showActionSheetWithOptions(
+      {
+        options: [...sortOptions.map(([, display]) => display), "Cancel"],
+        cancelButtonIndex,
+        userInterfaceStyle: theme.config.initialColorMode,
+      },
+      (index) => {
+        if (index === cancelButtonIndex) return;
+        const [key] = sortOptions[index];
+        feed.setSort(key);
+        onSortUpdate?.(key);
+      }
+    );
+  };
+  return <HeaderIconButton icon={SortIconType[feed.sort]} onPress={onPress} />;
```

Comment:
This is admittedly dangerous, but is basically the same logic we had before. Dangerous because `SortIconType` doesn't contain definitions for all the different `SortType`s, so it's possible for future selves to update sortOptions but not update SortIconType, which would result in... whatever happens when you try to wrap a `Pressable` around an `undefined`

Reply:
We can update some typescript stuff to help prevent that